sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -175,9 +181,24 @@ protected void initialize(String path, List<String> columns) throws IOException
         this.requestedSchema = ParquetSchemaConverter.EMPTY_MESSAGE();
       }
     }
-    reader.setRequestedSchema(requestedSchema);
+    fileReader.setRequestedSchema(requestedSchema);
     this.sparkSchema = new ParquetToSparkSchemaConverter(config).convert(requestedSchema);
-    this.totalRowCount = reader.getFilteredRecordCount();
+    this.totalRowCount = fileReader.getFilteredRecordCount();
+  }
+
+  protected void initialize(
```

Comment:
We can add a comment saying this is test only?

Reply:
marked it with `@VisibleForTesting`